src/mango/src/mango_selector.erl

```diff
@@ -398,6 +400,14 @@ negate({[{Field, Cond}]}) ->
     {[{Field, negate(Cond)}]}.
 
 
+norm_deduplicate({[{Op, [H|T]}]}) ->
+    Arg = [H | [X || X <- norm_deduplicate(T), X /= H]],
```

Comment:
This really hurts my brain to read. And not knowing what's supposed to call it I can't figure out if its correct.